app/serializers/user.js

```diff
@@ -5,9 +5,13 @@ export default ApplicationSerializer.extend({
   serialize(snapshot, options) {
     const json = this._super(...arguments);
     if (snapshot.id) {
-      let attributesToOmit = ['email'];
+      let attributesToOmit = [];
       if (!snapshot.adapterOptions || !snapshot.adapterOptions.includePassword) {
         attributesToOmit.push('password');
+      } else if (!snapshot.adapterOptions || !snapshot.adapterOptions.includeEmail) {
```

Comment:
Why are you making a POST request @sumedh123 ? POST is for register. The docs cannot get any clearer than this. 
- https://open-event-api.herokuapp.com/#users-user-details-patch 
- https://guides.emberjs.com/v2.14.0/models/creating-updating-and-deleting-records/#toc_persisting-records 

Thats the reason the API is expecting a password and an email. I donno how you're making a request here. save should be called on an existing user for ember to do a PATCH request.